backend/app/Application/Handlers/Issues/JoinIssueHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Handlers\Issues;
+
+use App\Application\Commands\Issues\JoinIssueCommand;
+use App\Application\Interfaces\WebSocketService;
+use App\Application\Services\CurrentUserService;
+use App\Domain\Entities\Issue;
+use App\Domain\Enums\IssueStatuses;
+use App\Domain\Enums\UserIssueStatuses;
+use App\Domain\Repositories\IssueRepository;
+
+class JoinIssueHandler
+{
+    public function __construct(
+        private CurrentUserService $currentUserService,
+        private IssueRepository $issueRepository,
+        private WebSocketService $webSocketService
+    ) { }
+    
+    public function handle(JoinIssueCommand $command): ?Issue
+    {
+        $issue = $this->issueRepository->findByNumber($command->getNumber());
+
+        if (!$issue) {
+            $issue = new Issue(
+                $command->getNumber(),
+                [
+                    $this->currentUserService->getUser()->getName()
+                ],
+                [
+                    [
+                        'user' => $this->currentUserService->getUser()->getName(),
+                        'status' => UserIssueStatuses::WAITING,
+                        'vote' => null
+                    ],
+                ],
+                IssueStatuses::VOTING
+            );
+        } else {
+            if (!in_array($this->currentUserService->getUser()->getName(), $issue->getUsers())){
```

Comment:
El identificador unívoco para un usuario es el nombre?

Reply:
Podés evitar el anidamiento de estos dos ifs poniendo directamente elseif (!in_array($this->currentUserService->getUser()->getName(), $issue->getUsers()))

Me queda la duda: Siempre el $this->currentUserService->getUser() va a dar no nulo? No estaría de más protegerse al comienzo de la operación con una aserción que valide que haya un usuario logueado

Reply:
Exacto, el identificador de un user es el name, y te rebota el register si es que queres tener dos users con el mismo name